packages/core/src/index.ts

```diff
@@ -1,3 +1,399 @@
+  _connectingWallet = undefined
+  // Cleanup WalletConnect QR if necessary since modal is now closed (URI set to
+  // undefined).
+  _onQrCloseCallback?.()
+  _onQrCloseCallback = undefined
+}
+
+// Connect WalletConnect client if necessary, and then connect to the wallet.
+export const connectToWallet = async (wallet: Wallet) => {
+  _connectingWallet = wallet
+  updateState({
+    status: CosmosWalletStatus.Connecting,
+    error: undefined,
+    displayingPicker: false,
+  })
+
+  let walletClient: unknown
+
+  // The actual meat of enabling and getting the wallet clients.
+  const finalizeWalletConnection = async (newWcSession?: boolean) => {
+    // Cleared in `cleanupAfterConnection`.
+    updateState({
+      enablingWallet: true,
+    })
+
+    const chainInfo = await getChainInfo(
+      _config.defaultChainId,
+      _config.chainInfoOverrides
+    )
+
+    walletClient = await wallet.getClient(
+      chainInfo,
+      _walletConnect,
+      newWcSession
+    )
+    if (!walletClient) {
+      throw new Error('Failed to retrieve wallet client.')
+    }
+
+    // Enable and connect to wallet, and retrieve data.
+    const connectedWallet = await getConnectedWalletInfo(
+      wallet,
+      walletClient,
+      chainInfo,
+      await _config.getSigningCosmWasmClientOptions?.(chainInfo),
+      await _config.getSigningStargateClientOptions?.(chainInfo)
+    )
+    // Add refresh listener to update connected wallet info.
+    wallet.addRefreshListener?.(_refreshListener)
+
+    // Allow to fail silently.
+    try {
+      await wallet.cleanupClient?.(walletClient)
+
+      // Save localStorage value for future autoconnection.
+      if (_config.localStorageKey) {
+        localStorage.setItem(_config.localStorageKey, wallet.id)
+      }
+    } catch (err) {
+      console.error(err)
+    }
+
+    updateState({
+      status: CosmosWalletStatus.Connected,
+      connectedWallet,
+    })
+  }
+
+  try {
+    // Connect to WalletConnect if necessary.
+    if (wallet.isWalletConnect) {
+      if (!wallet.walletConnectSigningMethods) {
+        throw new Error(
+          'Wallet definition missing WalletConnect signing methods.'
+        )
+      }
+
+      // Instantiate new WalletConnect instance if necessary.
+      if (!_walletConnect) {
+        _walletConnect = new (await import('@walletconnect/client')).default({
+          bridge: 'https://bridge.walletconnect.org',
+          signingMethods: wallet.walletConnectSigningMethods,
+          qrcodeModal: {
+            open: (walletConnectQrUri, closeCallback) => {
+              _onQrCloseCallback = closeCallback
+              updateState({
+                walletConnectQrUri,
+              })
+            },
+            // Occurs on disconnect, which is handled elsewhere.
+            close: () => console.log('qrcodeModal.close'),
+          },
+          // clientMeta,
+        })
+        // clientMeta in constructor is ignored for some reason, so
```

Comment:
this is weird, hope this doesnt break

Reply:
Yeah... found this hack in the Osmosis frontend.